src/libs/Geometry/geom_static.cpp

```diff
@@ -150,21 +162,20 @@ GEOM::GEOM(const char *fname, const char *lightname, GEOM_SERVICE &_srv, long fl
     }
     srv.free(rvb);
     // read vertices
-    auto *_colData = colData;
-    if (lightname != nullptr && flsz / 4 != nvertices)
+    if (colData.size() != nvertices)
     {
-        srv.free(colData);
-        lightname = nullptr;
+        colData.clear();
     }
+    auto _colData = colData.begin();
     for (v = 0; v < rhead.nvrtbuffs; v++)
     {
         auto *vrt = static_cast<RDF_VERTEX0 *>(srv.LockVertexBuffer(vbuff[v].dev_buff));
         srv.ReadFile(file, vrt, vbuff[v].size);
         for (long vr = 0; vr < vbuff[v].nverts; vr++)
         {
             auto *prv = (RDF_VERTEX0 *)((uint8_t *)(vrt) + vbuff[v].stride * vr);
-            if (lightname != nullptr)
-                prv->color = *_colData++;
+            if (!colData.empty())
+                prv->color = *(_colData++);
```

Comment:
This should be avoided if possible.
Shall be 
if (!colData.empty()) {
  prv->color = *_colData;
  ++_colData;
}